OSSM-8733 Add IstioRevisionTags to list of OSSM resources (#90659)
openshift/openshift-docs: modules/ossm-about-concepts-resources.adoc

Similarly, users create an `Istio` resource which instructs the {SMProduct} Operator to create a matching `IstioRevision` resource, which then in turn triggers the creation of the Istio control plane. To do that, the {SMProduct} Operator will copy all of your relevant configuration from the `Istio` resource to the `IstioRevision` resource.

[id="istiorevisiontag-resource_{context}"]
== IstioRevisionTag resource

> The `IstioRevisionTag` resource represents a stable revision tag, which functions as an alias for Istio control plane revisions. With a stable tag `prod`, you can use the label `istio.io/rev=prod` to inject proxies into your workloads. When you perform an upgrade to a control plane with a new revision name, you can update your tag to point to the new revision instead of having to re-label your workloads and namespaces. For more information, see "Stable Revision Tags."

Reviewer: ```suggestion
The `IstioRevisionTag` resource represents a stable revision tag, which functions as an alias for Istio control plane revisions. With the stable tag, `prod`, you can use the label `istio.io/rev=prod` to inject proxies into your workloads. When you perform an upgrade to a control plane with a new revision name, you can update your tag to point to the new revision instead of having to relabel your workloads and namespaces. For more information, see "Stable Revision Tags."
```
"relabel" is more used in American English (but "re-label" is not _wrong_)

> With the {SMProduct} Operator, you can use the `IstioRevisionTag` resource, which lets you reference both an `IstioRevision` and an `Istio` resource. When using an `Istio` resource and you update your control plane and the underlying `IstioRevision` changes, the {SMProduct} Operator updates your revision tag for you. You only need to restart your deployments to re-inject the new proxies.

Reviewer 2: ```suggestion
With the {SMProduct} Operator, you can use the `IstioRevisionTag` resource, which lets you reference both an `IstioRevision` and an `Istio` resource. When using an `Istio` resource, after you update your control plane the underlying `IstioRevision` changes and the {SMProduct} Operator updates your revision tag for you automatically. You only need to restart your deployments to re-inject the new proxies.
```

> With the {SMProduct} Operator, you can use the `IstioRevisionTag` resource, which lets you reference both an `IstioRevision` and an `Istio` resource. When using an `Istio` resource, after you update your control plane, the underlying `IstioRevision` changes, and the {SMProduct} Operator automatically updates your revision tag. You only need to restart your deployments to re-inject the new proxies.

Reviewer: ```suggestion
You can use the `IstioRevisionTag` resource with the {SMProduct} Operator, therefore you can reference both an `IstioRevision` and an `Istio` resource. When using an `Istio` resource, after you update your control plane, the underlying `IstioRevision` resource changes, and the {SMProduct} Operator automatically updates your revision tag. You only need to restart your deployments to re-inject the new proxies.
```
"lets you" is personification

> In Istio, stable revision tags are usually created using `istioctl` client. If you are using the {SMProduct} Operator, you can use the `IstioRevisionTag` resource, which lets you reference both an `IstioRevision` and an `Istio` resource. Now, when you update your control plane and the underlying `IstioRevision` changes, the {SMProduct} Operator updates your revision tag for you. You only need to restart your deployments to re-inject the new proxies.

Reviewer 2: "Now, when you update your control plane..." this applies only when referencing `Istio` resource, it should be described clearly here.

> In Istio, stable revision tags are usually created using `istioctl` client. If you are using the {SMProduct} Operator, you can use the `IstioRevisionTag` resource, which lets you reference both an `IstioRevision` and an `Istio` resource. When using an `Istio` resource, when you update your control plane and the underlying `IstioRevision` changes, the {SMProduct} Operator updates your revision tag for you. You only need to restart your deployments to re-inject the new proxies.

Reviewer 3: tiny nit: pls feel free to ignore this comment if its not required.

```suggestion
In Istio, stable revision tags are usually created using `istioctl` client binary. If you are using the {SMProduct} Operator, you can use the `IstioRevisionTag` resource, which lets you reference both an `IstioRevision` and an `Istio` resource. When using an `Istio` resource, when you update your control plane and the underlying `IstioRevision` changes, the {SMProduct} Operator updates your revision tag for you. You only need to restart your deployments to re-inject the new proxies.
```

> In Istio, stable revision tags are usually created using `istioctl` client binary. If you are using the {SMProduct} Operator, you can use the `IstioRevisionTag` resource, which lets you reference both an `IstioRevision` and an `Istio` resource. When using an `Istio` resource, when you update your control plane and the underlying `IstioRevision` changes, the {SMProduct} Operator updates your revision tag for you. You only need to restart your deployments to re-inject the new proxies.

Reviewer 3: > In Istio, stable revision tags are usually created using `istioctl` client binary

@dgn @FilipB do we support creating the revision tags using the `istioctl` binary in OSSM? If not, we should probably add a corresponding note.

Author: This is the source: https://github.com/istio-ecosystem/sail-operator/blob/main/docs/README.md#istiorevisiontag-resource 

References to `istioctl` say "client", and say "binary" when directing users to install it.

Reviewer 3: > In Istio, stable revision tags are usually created using istioctl, but if you're using the Sail Operator, you can use the IstioRevisionTag resource, which comes with an additional feature: instead of just being able to reference an IstioRevision, you can also reference an Istio resource.

From the upstream document, I see that we support tags using the `istioctl` binary as well.

Reviewer 2: I believe we don't want users to use istioctl to create tags. Looking at https://docs.redhat.com/en/documentation/red_hat_openshift_service_mesh/3.0/html/installing/ossm-istioctl-tool#ossm-istioctl-tool the `tag` command is not supported. I agree that `In Istio, stable revision tags are usually created using istioctl, but if you're using the Sail Operator` might be confusing and we should either not talk about upstream istioctl for tagging at all or say not to use it.

Author: @FilipB updated. Place take a look.

.Example `IstioRevisionTag` resource
[source,yaml]
----
apiVersion: sailoperator.io/v1
kind: IstioRevisionTag
metadata:
  name: default
spec:
  targetRef:
    kind: Istio <1>
    name: prod  <2>
----
<1> This can be either `Istio` or `IstioRevision`.
<2> The name of the `Istio` or `IstioRevision` resource.

> The `IstioRevisionTag` has one field in its spec: `targetRef`, which can reference an `Istio` or `IstioRevision` resource. After deploying the `IstioRevisionTag`, you can use both the `istio.io/rev=default` and also `istio-injection=enabled` labels to inject proxies into your workloads.

Reviewer: ```suggestion
The `IstioRevisionTag` has one field in its `spec:`, `targetRef`, which can reference an `Istio` or `IstioRevision` resource. After deploying the `IstioRevisionTag`, you can use both the `istio.io/rev=default` and `istio-injection=enabled` labels to inject proxies into your workloads.
```

> The `istio-injection` label can only be used for revisions and revision tags named `default`, like the `IstioRevisionTag` provided in the example.

Reviewer: Should this be an [IMPORTANT] admonition?

Similarly, users create an `Istio` resource which instructs the {SMProduct} Operator to create a matching `IstioRevision` resource, which then in turn triggers the creation of the Istio control plane. To do that, the {SMProduct} Operator will copy all of your relevant configuration from the `Istio` resource to the `IstioRevision` resource.

[id="istiorevisiontag-resource_{context}"]
== IstioRevisionTag resource

> The `IstioRevisionTag` resource represents a stable revision tag, that functions as an alias for Istio control plane revisions. With the stable tag, `prod`, you can use the label `istio.io/rev=prod` to inject proxies into your workloads. When you perform an upgrade to a control plane with a new revision name, you can update your tag to point to the new revision instead of having to relabel your workloads and namespaces. For more information, see "Stable Revision Tags."

Reviewer 4: ```suggestion
The `IstioRevisionTag` resource represents a stable revision tag that functions as an alias for Istio control plane revisions. With the stable tag, `prod`, you can use the label `istio.io/rev=prod` to inject proxies into your workloads. When you perform an upgrade to a control plane with a new revision name, you can update your tag to point to the new revision instead of having to relabel your workloads and namespaces. For more information, see "Stable Revision Tags."
```

> The `IstioRevisionTag` resource represents a stable revision tag, which functions as an alias for Istio control plane revisions. With the stable tag, `prod`, you can use the label `istio.io/rev=prod` to inject proxies into your workloads. When you perform an upgrade to a control plane with a new revision name, you can update your tag to point to the new revision instead of having to relabel your workloads and namespaces. For more information, see "Stable Revision Tags."

Reviewer 4: ```suggestion
The `IstioRevisionTag` resource represents a stable revision tag that functions as an alias for Istio control plane revisions. With the stable tag, `prod`, you can use the label `istio.io/rev=prod` to inject proxies into your workloads. When you perform an upgrade to a control plane with a new revision name, you can update your tag to point to the new revision instead of having to relabel your workloads and namespaces. For more information, see "Stable Revision Tags."
```

Is this maximally specific in a restrictive clause sense?

> You can use the `IstioRevisionTag` resource with the {SMProduct} Operator, therefore you can reference both an `IstioRevision` and an `Istio` resource. When using an `Istio` resource, after you update your control plane, the underlying `IstioRevision` resource changes, and the {SMProduct} Operator automatically updates your revision tag. You only need to restart your deployments to re-inject the new proxies.

Reviewer 4: If you feel like it:
```suggestion
You can use the `IstioRevisionTag` resource with the {SMProduct} Operator. Therefore, you can reference both an `IstioRevision` and an `Istio` resource. When using an `Istio` resource, after you update your control plane, the underlying `IstioRevision` resource changes, and the {SMProduct} Operator automatically updates your revision tag. You only need to restart your deployments to re-inject the new proxies.
```

> The `IstioRevisionTag` has one field in its `spec:`, `targetRef`, which can reference an `Istio` or `IstioRevision` resource. After deploying the `IstioRevisionTag`, you can use both the `istio.io/rev=default` and `istio-injection=enabled` labels to inject proxies into your workloads.

Reviewer 4: Is there a noun you could pair with "`spec:`"?

[Important]
====

> The `istio-injection` label can only be used for revisions and revision tags named `default`, like the `IstioRevisionTag` resource in the following example:

Reviewer 4: ```suggestion
The `istio-injection` label can only be used for revisions and revision tags that have the name `default`, like the `IstioRevisionTag` resource in the following example:
```

.Example `IstioRevisionTag` resource
[source,yaml]
----
apiVersion: sailoperator.io/v1
kind: IstioRevisionTag
metadata:
  name: default
spec:
  targetRef:
    kind: Istio <1>
    name: prod  <2>
----

> <1> This can be either `Istio` or `IstioRevision`.

Reviewer 4: ```suggestion
<1> This value can be either `Istio` or `IstioRevision`.
```

      - kube-system
----

[role="_additional-resources"]
[id="additional-resources_{context}"]
== Additional resources

> * link:https://istio.io/latest/docs/setup/upgrade/canary/#stable-revision-labels[Stable Revision Tags] (Istio documentation)

Reviewer 4: Is there a way to rewrite this to avoid the parens?

Reviewer 4: > In a few older PRs, since it links to a non-Red Hat doc, reviewers have said to put the source in parenthesis.

I don't think that's correct. There's nothing in ISG, SSG, or the repo guidance about using parens to call out external sites. There is a carve-out specifically for KB pages in the SSG, though.

Reviewer 4: I do think the current format has to go. For additional resources in the repo guidance:

> You must not include text outside of the square brackets. Text outside the brackets does not effectively render in an additional resources section box for a document on the Customer Portal.

https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#additional-resources-sections

Author: I was sent this: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#lead-in-link-sentence-style-for-the-link-attribute

"When pointing to a link that exists external to Red Hat’s customer portal documentation, add the link attribute, and use the <document_heading_name> (<document_source>)"

Author: Jira to revisit and address https://issues.redhat.com/browse/OSSM-9098
